runtime/fastly/builtins/fetch-event.cpp

```diff
@@ -24,6 +29,283 @@ namespace fastly::fetch_event {
 
 namespace {
 
+JSString *address(JSObject *obj) {
+  JS::Value val = JS::GetReservedSlot(obj, static_cast<uint32_t>(ClientInfo::Slots::Address));
+  return val.isString() ? val.toString() : nullptr;
+}
```

Comment:
Why inline the ClientInfo implementation instead of copying over individual files from the existing builtin?

Reply:
The argument here is that clientInfo isn't its own builtin as it is part of the "fastly fetch event" at this point.

Perhaps in future we could refactor our fetch event to be shared with StarlingMonkey in future, and treat the Fastly-specific stuff as monkey patching on top, but even then treating each feature as its own builtin may likely be overkill, and instead we may still have a single "fastly" builtin that does the Fastly patches.